### PR TITLE
Network Area Diagrams: add initial positions to metadata

### PR DESCRIPTION
**Pull request by zamarrenolm**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Hidden nodes do not have positions in the SVG.

**What is the new behavior (if this is a feature change)?**
We add initial positions for all nodes in the metadata.

### CODE REVIEW

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java:6]: For new files we now put the SPDX identifier to ease license scanning:
```java
/**
 * Copyright (c) 2022, RTE (http://www.rte-france.com/)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 * SPDX-License-Identifier: MPL-2.0
 */
```

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            XMLStreamWriter writer) throws XMLStreamException {
+        if (items.isEmpty()) {
+            writer.writeEmptyElement(METADATA_PREFIX, collectionElementName, METADATA_NAMESPACE_URI);
+        } else {
+            writer.writeStartElement(METADATA_PREFIX, collectionElementName, METADATA_NAMESPACE_URI);
+            for (I item : items) {
+                writer.writeEmptyElement(METADATA_PREFIX, itemElementName, METADATA_NAMESPACE_URI);
+                writer.writeAttribute(DIAGRAM_ID_ATTRIBUTE, diagramIdToSvgId.apply(item.getIdentifiable().getDiagramId()));
+                writer.writeAttribute(EQUIPMENT_ID_ATTRIBUTE, item.getIdentifiable().getEquipmentId());
+                elemWriter.write(item, writer);
```
Comment (flo-dup) [inline network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java:176]: You don't need a new interface `MetadataItemWriter`, you could use directly `IdentifiableMetadata`:
```java
item.write(writer);
```
Then you need to add `getElementName` to have a default implementation in `IdentifiableMetadata`, and a `svgId` attribute in it also

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java (hunk 1 above)
Comment (zamarrenolm) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java]: updated

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java (hunk 2 above)
Comment (zamarrenolm) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/DiagramMetadata.java]: refactored the metadata items so they know how to write themselves to xml. Also refactored the reads. The diagram metadata only knows about serialization/deserialization of the collections it owns.

**Review by flo-dup — CHANGES_REQUESTED** [review]

File: network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/NodeMetadata.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ */
+public class NodeMetadata extends AbstractMetadataItem {
+    private final Point position;
+
+    private static final String ELEMENT_NAME = "node";
+    private static final String BUS_NODE_ELEMENT_NAME = "busNode";
+    private static final String POSITION_X_ATTRIBUTE = "x";
+    private static final String POSITION_Y_ATTRIBUTE = "y";
+    private static final String POSITION_COORD_FORMAT = "%.4f";
```
[the remark this answers is not in the document]
Comment (zamarrenolm) [reply network-area-diagram/src/main/java/com/powsybl/nad/svg/metadata/NodeMetadata.java]: done

**Review by flo-dup — APPROVED** [review]

**Pull request merged**